pkg/yang/types_builtin.go

```diff
@@ -696,12 +714,8 @@ func (n Number) Int() (int64, error) {
 	return 0, errors.New("unknown number type")
 }
 
-// add adds i to n without checking overflow.  We really only need to be
-// able to add 1 for our code. panics if n is a decimal.
-func (n Number) add(i uint64) Number {
-	if n.IsDecimal() {
-		panic("cannot call add() on decimal number " + n.String())
-	}
+// addQuantum adds the smallest quantum to n without checking overflow.
+func (n Number) addQuantum(i uint64) Number {
```

Comment:
What do you mean by quantum here? I'd read this as meaning that we add the smallest increment that is possible in the number (e.g., `fraction-digits` of 1 would mean that we add 0.1), but then the `uint64` argument doesn't make sense.

Reply:
`addQuantum`'s only purpose right now is to support coalescing by incrementing the max of the previous range to see whether it overlaps with the next range. 

The fractional digit
```
controls the size of the minimum difference between values of a
   decimal64 type by restricting the value space to numbers that are
   expressible as "i x 10^-n" where n is the fraction-digits argument.
```
So, coalescing decimals is akin to coalescing integers after multiplying by 10^n.